.github/workflows/manual-publish.yml

```diff
@@ -0,0 +1,50 @@
+name: Publish Package
+on:
+  workflow_dispatch:
+    inputs:
+      dry_run:
+        description: 'Is this a dry run. If so no package will be published.'
+        type: boolean
+        required: true
+
+jobs:
+  build-publish:
+    runs-on: ubuntu-latest
+    # Needed to get tokens during publishing.
+    permissions:
+      id-token: write
+      contents: read
+    outputs:
+      gem-hash: ${{ steps.publish.outputs.gem-hash}}
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: launchdarkly/gh-actions/actions/release-secrets@release-secrets-v1.0.0
+        name: 'Get rubygems API key'
+        with:
+          aws_assume_role: ${{ vars.AWS_ROLE_ARN }}
+          ssm_parameter_pairs: '/production/common/releasing/rubygems/api_key = GEM_HOST_API_KEY'
+
+      - id: build-and-test
+        name: Build and Test
+        uses: ./.github/actions/ci
+        with:
+          ruby-version: 3.0
+
+      - id: publish
+        name: Publish Package
+        uses: ./.github/actions/publish
+        with:
+          dry_run: ${{ inputs.dry_run }}
+
+  release-provenance:
+    needs: [ 'build-publish' ]
+    permissions:
+      actions: read
+      id-token: write
+      contents: write
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.7.0
+    with:
+      base64-subjects: "${{ needs.build-publish.outputs.gem-hash }}"
+      upload-assets: ${{ !inputs.dry_run }}
+        
```

Comment:
I wonder if this last line has a space or something. I get :no_entry: in my diff.

Reply:
Yeah, trailing line with spaces. Cleaned it up.